internal/backends/nodejs/grab.go

```diff
@@ -105,24 +106,18 @@ func parseFile(contents []byte, results chan parseResult) {
 		}
 
 		var importPath string
-
-		if match.PatternIndex == 0 {
-			importPath = match.Captures[0].Node.Content(contents)
-		} else if match.PatternIndex == 1 {
-			// TODO: https://github.com/smacker/go-tree-sitter/issues/110
-			// once the above issue is resolved, uncomment the `@other-imports` predicate
-			// and remove this check
-			if len(match.Captures) != 2 {
-				continue
+		for _, capture := range match.Captures {
+			if query.CaptureNameForId(capture.Index) == "import" {
+				importPath = capture.Node.Content(contents)
+				break
 			}
+		}
 
-			importPath = match.Captures[1].Node.Content(contents)
+		if importPath == "" {
+			// shouldn't happen, with the way the query is written and handled.
+			util.Die("Failed to parse import path")
```

Comment:
I feel like we don't have enough observability to react to these effectively, but maybe we fixed that

Reply:
@ryantm has a WIP branch adding o11y to upm